src/features/auth/pages/Signup.test.tsx

```diff
@@ -33,39 +31,54 @@ describe('<Signup />', () => {
   })
 
   it('should redirect to SignUpSignInChoiceModal when clicking on ArrowPrevious icon', async () => {
+    const {
+      result: {
+        current: { navigate },
+      },
+    } = renderHook(() => useNavigation())
+    navigate.mockReset()
     const { getByTestId } = renderPage()
 
     const leftIcon = getByTestId('leftIcon')
     fireEvent.press(leftIcon)
 
-    await waitForExpect(() => {
-      expect(navigationTestProps.navigation.navigate).toBeCalledTimes(1)
-      expect(navigationTestProps.navigation.navigate).toHaveBeenCalledWith('Home', {
-        shouldDisplayLoginModal: true,
-      })
+    expect(navigate).toBeCalledWith('Home', {
+      shouldDisplayLoginModal: true,
     })
   })
 
-  it('should navigate to ChoosePassword page when submit email', async () => {
-    const { getByPlaceholderText, findByText } = renderPage()
+  it('should redirect to ChoosePassword on valid email', () => {
+    const {
+      result: {
+        current: { navigate },
+      },
+    } = renderHook(() => useNavigation())
+    navigate.mockReset()
+
+    const { getByText, getByPlaceholderText, queryByText } = renderPage()
 
     const emailInput = getByPlaceholderText('tonadresse@email.com')
     fireEvent.changeText(emailInput, 'john.doe@gmail.com')
 
-    const continueButton = await findByText('Continuer')
+    const continueButton = getByText('Continuer')
     fireEvent.press(continueButton)
 
-    await waitForExpect(() => {
-      expect(navigationTestProps.navigation.navigate).toBeCalledTimes(1)
-      expect(navigationTestProps.navigation.navigate).toHaveBeenCalledWith('ChoosePassword')
-    })
+    expect(navigate).toBeCalledWith('ChoosePassword')
+    expect(queryByText("Format de l'e-mail incorrect")).toBeFalsy()
+  })
+  it('should reject email', () => {
+    const { getByText, getByPlaceholderText, queryByText } = renderPage()
+
+    const emailInput = getByPlaceholderText('tonadresse@email.com')
+    fireEvent.changeText(emailInput, 'john.doe')
+
+    const continueButton = getByText('Continuer')
+    fireEvent.press(continueButton)
+
+    expect(queryByText("Format de l'e-mail incorrect")).toBeTruthy()
```

Comment:
Je propose d'ajouter `expect(navigate).not.toHaveBeenCalled()`, ainsi nous sommes sûr que le parcours s'est arrêté